extension/partiql-extension-ion/src/boxed_ion.rs

```diff
@@ -0,0 +1,587 @@
+        todo!()
+    }
+}
+
+impl TupleDatum for BoxedIon {
+    fn len(&self) -> usize {
+        match &self.doc {
+            BoxedIonValue::Stream() => {
+                todo!()
+            }
+            BoxedIonValue::Sequence(_) => {
+                todo!()
+            }
+            BoxedIonValue::Value(elt) => match elt.expect_struct() {
+                Ok(strct) => strct.len(),
+                Err(_) => todo!(),
+            },
+        }
+    }
+}
+
+impl<'a> RefTupleView<'a, Value> for BoxedIon {
+    fn get_val(&self, target_key: &BindingsName<'_>) -> Option<Cow<'a, Value>> {
+        let matcher = target_key.matcher();
+        let Self { doc, ctx } = self;
+        match doc {
+            BoxedIonValue::Stream() => {
+                todo!()
+            }
+            BoxedIonValue::Sequence(_) => {
+                todo!()
+            }
+            BoxedIonValue::Value(elt) => match elt.expect_struct() {
+                Ok(strct) => {
+                    for (k, elt) in strct {
+                        if let Some(k) = k.text() {
+                            if matcher.matches(k) {
+                                return Some(Cow::Owned(Self::new_value(elt.clone(), ctx.clone())));
+                            }
+                        }
+                    }
+                    None
+                }
+                Err(_) => todo!(),
+            },
+        }
+    }
+}
+
+impl OwnedTupleView<Value> for BoxedIon {
+    fn take_val(self, k: &BindingsName<'_>) -> Option<Value> {
+        todo!()
+    }
+
+    fn take_val_boxed(self: Box<Self>, target_key: &BindingsName<'_>) -> Option<Value> {
+        let matcher = target_key.matcher();
+        let Self { doc, ctx } = *self;
+        match doc {
+            BoxedIonValue::Stream() => {
+                todo!()
+            }
+            BoxedIonValue::Sequence(_) => {
+                todo!()
+            }
+            BoxedIonValue::Value(elt) => match elt.try_into_struct() {
+                Ok(strct) => {
+                    for (k, elt) in strct {
+                        if let Some(k) = k.text() {
+                            if matcher.matches(k) {
+                                return Some(Self::new_value(elt, ctx));
+                            }
+                        }
+                    }
+                    None
+                }
+                Err(_) => todo!(),
+            },
+        }
+    }
+}
+
+impl Debug for BoxedIon {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("BoxedIon").field(&self.doc).finish()
+    }
+}
+
+impl Display for BoxedIon {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&self.doc, f)
+    }
+}
+
+impl BoxedIon {
+    fn into_value(self) -> Value {
+        Value::from(Variant::from(self))
+    }
+
+    pub fn new(doc: impl Into<BoxedIonValue>, ctx: IonContextPtr) -> Self {
```

Comment:
Some clippy warnings in this file for `...is more private than the item...`

Reply:
Fixed in #544 : for example: https://github.com/partiql/partiql-lang-rust/pull/544/files#diff-b95d0de35899342f30d2531c8cd3728a958d846d163f0c154c204f7aeb8979ceL112